app/controllers/photos_controller.rb

```diff
@@ -0,0 +1,55 @@
+class PhotosController < ApplicationController
+  before_action :authenticate_user!, :except => [:index, :show]
+
+  def index
+    @photos = Photo.all
+  end
+
+  def new
+    @photo = Photo.new
+  end
+
+  def create
+    @photo = Photo.new( photo_params )
+    if @photo.save
+      redirect_to photos_path
+    else
+      render 'new'
+    end
+  end
+
+  def show
+    @photo = Photo.find(params[:id])
+  end
+
+  def edit
+    @photo = Photo.find(params[:id])
```

Comment:
`@photo = Photo.find(params[:id])` is repeated 3 times. This could be refactored to a before_action.